This patch disable the scroll lock keypresses that were added to 0.99.4 for the 
screensaver disabling, reverting to the old behaviour and fixing the prolems 
with terminal emulators. See bug #102884.
Index: xine-ui-0.99.4/src/xitk/videowin.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/videowin.c
+++ xine-ui-0.99.4/src/xitk/videowin.c
@@ -1449,8 +1449,8 @@ void video_window_init (window_attribute
   gVw->desktopHeight      = DisplayHeight(gGui->video_display, gGui->video_screen);
 
 #ifdef HAVE_XTESTEXTENSION
-  gVw->fake_keys[0] = XKeysymToKeycode(gGui->video_display, XK_Scroll_Lock);
-  gVw->fake_keys[1] = XKeysymToKeycode(gGui->video_display, XK_Num_Lock);
+  gVw->fake_keys[0] = XKeysymToKeycode(gGui->video_display, XK_Shift_L);
+  gVw->fake_keys[1] = XKeysymToKeycode(gGui->video_display, XK_Control_L);
   gVw->fake_key_cur = 0;
 #endif
   
@@ -2151,8 +2151,6 @@ void video_window_reset_ssaver(void) {
       XLockDisplay(gGui->video_display);
       XTestFakeKeyEvent(gGui->video_display, gVw->fake_keys[gVw->fake_key_cur], True, CurrentTime);
       XTestFakeKeyEvent(gGui->video_display, gVw->fake_keys[gVw->fake_key_cur], False, CurrentTime);
-      XTestFakeKeyEvent(gGui->video_display, gVw->fake_keys[gVw->fake_key_cur], True, CurrentTime);
-      XTestFakeKeyEvent(gGui->video_display, gVw->fake_keys[gVw->fake_key_cur], False, CurrentTime);
       XSync(gGui->video_display, False);
       XUnlockDisplay(gGui->video_display);
     }
Index: xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/xine-toolkit/xitk.c
+++ xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.c
@@ -147,7 +147,7 @@ typedef struct {
   
   struct timeval              keypress;
 
-  KeyCode                     ignore_keys[3];
+  KeyCode                     ignore_keys[2];
 
   pthread_t                  *tips_thread;
   unsigned long               tips_timeout;
@@ -1803,9 +1803,8 @@ void xitk_init(Display *display, XColor 
   xitk_x_error           = 0;
   gXitk->x_error_handler = NULL;
   gXitk->modalw          = None;
-  gXitk->ignore_keys[0]  = XKeysymToKeycode(display, XK_Scroll_Lock);
-  gXitk->ignore_keys[1]  = XKeysymToKeycode(display, XK_Num_Lock);
-  gXitk->ignore_keys[2]  = XKeysymToKeycode(display, XK_Caps_Lock);
+  gXitk->ignore_keys[0]  = XKeysymToKeycode(display, XK_Shift_L);
+  gXitk->ignore_keys[1]  = XKeysymToKeycode(display, XK_Control_L);
   gXitk->tips_timeout    = TIPS_TIMEOUT;
   XGetInputFocus(display, &(gXitk->parent.window), &(gXitk->parent.focus));
 
